#DataFrames: Using Julia

## DataFrames package is required to read any data frame

Install DataFrames Package in Julia

In [3]:
Pkg.add("DataFrames")


INFO: Cloning cache of ArrayViews from git://github.com/JuliaLang/ArrayViews.jl.git
INFO: Cloning cache of DataArrays from git://github.com/JuliaStats/DataArrays.jl.git
INFO: Cloning cache of DataFrames from git://github.com/JuliaStats/DataFrames.jl.git
INFO: Cloning cache of Docile from git://github.com/MichaelHatherly/Docile.jl.git
INFO: Cloning cache of GZip from git://github.com/JuliaLang/GZip.jl.git
INFO: Cloning cache of Reexport from git://github.com/simonster/Reexport.jl.git
INFO: Cloning cache of SortingAlgorithms from git://github.com/JuliaLang/SortingAlgorithms.jl.git
INFO: Cloning cache of StatsBase from git://github.com/JuliaStats/StatsBase.jl.git
INFO: Cloning cache of StatsFuns from git://github.com/JuliaStats/StatsFuns.jl.git
INFO: Installing ArrayViews v0.6.4
INFO: Installing DataArrays v0.2.20
INFO: Installing DataFrames v0.6.10
INFO: Installing Docile v0.5.19
INFO: Installing GZip v0.2.18
INFO: Installing Reexport v0.0.3
INFO: Installing SortingAlgorithms v0.0.6
INFO

Declear DataFrames package

In [1]:
using DataFrames

##Read data from csv file

In [2]:
df = readtable("C:/Users/JB/home_data.csv")

21613x21 DataFrame
| Row   | id         | date              | price   | bedrooms | bathrooms |
|-------|------------|-------------------|---------|----------|-----------|
| 1     | 7129300520 | "20141013T000000" | 221900  | 3        | 1.0       |
| 2     | 6414100192 | "20141209T000000" | 538000  | 3        | 2.25      |
| 3     | 5631500400 | "20150225T000000" | 180000  | 2        | 1.0       |
| 4     | 2487200875 | "20141209T000000" | 604000  | 4        | 3.0       |
| 5     | 1954400510 | "20150218T000000" | 510000  | 3        | 2.0       |
| 6     | 7237550310 | "20140512T000000" | 1225000 | 4        | 4.5       |
| 7     | 1321400060 | "20140627T000000" | 257500  | 3        | 2.25      |
| 8     | 2008000270 | "20150115T000000" | 291850  | 3        | 1.5       |
| 9     | 2414600126 | "20150415T000000" | 229500  | 3        | 1.0       |
| 10    | 3793500160 | "20150312T000000" | 323000  | 3        | 2.5       |
| 11    | 1736800520 | "20150403T000000" | 662500  | 3        | 2.5       |
⋮
| 21602 | 5100403806 | "20150407T000000" | 467000  | 3        | 2.5       |
| 21603 | 844000965  | "20140626T000000" | 224000  | 3        | 1.75      |
| 21604 | 7852140040 | "20140825T000000" | 507250  | 3        | 2.5       |
| 21605 | 9834201367 | "20150126T000000" | 429000  | 3        | 2.0       |
| 21606 | 3448900210 | "20141014T000000" | 610685  | 4        | 2.5       |
| 21607 | 7936000429 | "20150326T000000" | 1007500 | 4        | 3.5       |
| 21608 | 2997800021 | "20150219T000000" | 475000  | 3        | 2.5       |
| 21609 | 263000018  | "20140521T000000" | 360000  | 3        | 2.5       |
| 21610 | 6600060120 | "20150223T000000" | 400000  | 4        | 2.5       |
| 21611 | 1523300141 | "20140623T000000" | 402101  | 2        | 0.75      |
| 21612 | 291310100  | "20150116T000000" | 400000  | 3        | 2.5       |
| 21613 | 1523300157 | "20141015T000000" | 325000  | 2        | 0.75      |

| Row   | sqft_living | sqft_lot | floors | waterfront | view | condition |
|-------|-------------|----------|--------|------------|------|-----------|
| 1     | 1180        | 5650     | 1.0    | 0          | 0    | 3         |
| 2     | 2570        | 7242     | 2.0    | 0          | 0    | 3         |
| 3     | 770         | 10000    | 1.0    | 0          | 0    | 3         |
| 4     | 1960        | 5000     | 1.0    | 0          | 0    | 5         |
| 5     | 1680        | 8080     | 1.0    | 0          | 0    | 3         |
| 6     | 5420        | 101930   | 1.0    | 0          | 0    | 3         |
| 7     | 1715        | 6819     | 2.0    | 0          | 0    | 3         |
| 8     | 1060        | 9711     | 1.0    | 0          | 0    | 3         |
| 9     | 1780        | 7470     | 1.0    | 0          | 0    | 3         |
| 10    | 1890        | 6560     | 2.0    | 0          | 0    | 3         |
| 11    | 3560        | 9796     | 1.0    | 0          | 0    | 3         |
⋮
| 21602 | 1425        | 1179     | 3.0    | 0          | 0    | 3         |
| 21603 | 1500        | 11968    | 1.0    | 0          | 0    | 3         |
| 21604 | 2270        | 5536     | 2.0    | 0          | 0    | 3         |
| 21605 | 1490        | 1126     | 3.0    | 0          | 0    | 3         |
| 21606 | 2520        | 6023     | 2.0    | 0          | 0    | 3         |
| 21607 | 3510        | 7200     | 2.0    | 0          | 0    | 3         |
| 21608 | 1310        | 1294     | 2.0    | 0          | 0    | 3         |
| 21609 | 1530        | 1131     | 3.0    | 0          | 0    | 3         |
| 21610 | 2310        | 5813     | 2.0    | 0          | 0    | 3         |
| 21611 | 1020        | 1350     | 2.0    | 0          | 0    | 3         |
| 21612 | 1600        | 2388     | 2.0    | 0          | 0    | 3         |
| 21613 | 1020        | 1076     | 2.0    | 0          | 0    | 3         |

| Row   | grade | sqft_above | sqft_basement | yr_built | yr_renovated |
|-------|-------|------------|---------------|----------|--------------|
| 1     | 7     | 1180       

###Print top 6 rows of the DataFrame

In [3]:
head(df)

6x21 DataFrame
| Row | id         | date              | price   | bedrooms | bathrooms |
|-----|------------|-------------------|---------|----------|-----------|
| 1   | 7129300520 | "20141013T000000" | 221900  | 3        | 1.0       |
| 2   | 6414100192 | "20141209T000000" | 538000  | 3        | 2.25      |
| 3   | 5631500400 | "20150225T000000" | 180000  | 2        | 1.0       |
| 4   | 2487200875 | "20141209T000000" | 604000  | 4        | 3.0       |
| 5   | 1954400510 | "20150218T000000" | 510000  | 3        | 2.0       |
| 6   | 7237550310 | "20140512T000000" | 1225000 | 4        | 4.5       |

| Row | sqft_living | sqft_lot | floors | waterfront | view | condition |
|-----|-------------|----------|--------|------------|------|-----------|
| 1   | 1180        | 5650     | 1.0    | 0          | 0    | 3         |
| 2   | 2570        | 7242     | 2.0    | 0          | 0    | 3         |
| 3   | 770         | 10000    | 1.0    | 0          | 0    | 3         |
| 4   | 1960        | 5000     | 1.0    | 0          | 0    | 5         |
| 5   | 1680        | 8080     | 1.0    | 0          | 0    | 3         |
| 6   | 5420        | 101930   | 1.0    | 0          | 0    | 3         |

| Row | grade | sqft_above | sqft_basement | yr_built | yr_renovated | zipcode |
|-----|-------|------------|---------------|----------|--------------|---------|
| 1   | 7     | 1180       | 0             | 1955     | 0            | 98178   |
| 2   | 7     | 2170       | 400           | 1951     | 1991         | 98125   |
| 3   | 6     | 770        | 0             | 1933     | 0            | 98028   |
| 4   | 7     | 1050       | 910           | 1965     | 0            | 98136   |
| 5   | 8     | 1680       | 0             | 1987     | 0            | 98074   |
| 6   | 11    | 3890       | 1530          | 2001     | 0            | 98053   |

| Row | lat     | long     | sqft_living15 | sqft_lot15 |
|-----|---------|----------|---------------|------------|
| 1   | 47.5112 | -122.257 | 1340          | 5650       |
| 2   | 47.721  | -122.319 | 1690          | 7639       |
| 3   | 47.7379 | -122.233 | 2720          | 8062       |
| 4   | 47.5208 | -122.393 | 1360          | 5000       |
| 5   | 47.6168 | -122.045 | 1800          | 7503       |
| 6   | 47.6561 | -122.005 | 4760          | 101930     |

###Print bottom 6 rows of the DataFrame

In [4]:
tail(df)

6x21 DataFrame
| Row | id         | date              | price  | bedrooms | bathrooms |
|-----|------------|-------------------|--------|----------|-----------|
| 1   | 2997800021 | "20150219T000000" | 475000 | 3        | 2.5       |
| 2   | 263000018  | "20140521T000000" | 360000 | 3        | 2.5       |
| 3   | 6600060120 | "20150223T000000" | 400000 | 4        | 2.5       |
| 4   | 1523300141 | "20140623T000000" | 402101 | 2        | 0.75      |
| 5   | 291310100  | "20150116T000000" | 400000 | 3        | 2.5       |
| 6   | 1523300157 | "20141015T000000" | 325000 | 2        | 0.75      |

| Row | sqft_living | sqft_lot | floors | waterfront | view | condition |
|-----|-------------|----------|--------|------------|------|-----------|
| 1   | 1310        | 1294     | 2.0    | 0          | 0    | 3         |
| 2   | 1530        | 1131     | 3.0    | 0          | 0    | 3         |
| 3   | 2310        | 5813     | 2.0    | 0          | 0    | 3         |
| 4   | 1020        | 1350     | 2.0    | 0          | 0    | 3         |
| 5   | 1600        | 2388     | 2.0    | 0          | 0    | 3         |
| 6   | 1020        | 1076     | 2.0    | 0          | 0    | 3         |

| Row | grade | sqft_above | sqft_basement | yr_built | yr_renovated | zipcode |
|-----|-------|------------|---------------|----------|--------------|---------|
| 1   | 8     | 1180       | 130           | 2008     | 0            | 98116   |
| 2   | 8     | 1530       | 0             | 2009     | 0            | 98103   |
| 3   | 8     | 2310       | 0             | 2014     | 0            | 98146   |
| 4   | 7     | 1020       | 0             | 2009     | 0            | 98144   |
| 5   | 8     | 1600       | 0             | 2004     | 0            | 98027   |
| 6   | 7     | 1020       | 0             | 2008     | 0            | 98144   |

| Row | lat     | long     | sqft_living15 | sqft_lot15 |
|-----|---------|----------|---------------|------------|
| 1   | 47.5773 | -122.409 | 1330          | 1265       |
| 2   | 47.6993 | -122.346 | 1530          | 1509       |
| 3   | 47.5107 | -122.362 | 1830          | 7200       |
| 4   | 47.5944 | -122.299 | 1020          | 2007       |
| 5   | 47.5345 | -122.069 | 1410          | 1287       |
| 6   | 47.5941 | -122.299 | 1020          | 1357       |

###Find the dimension of DataFrame

In [5]:
size(df)

(21613,21)

size prints DataFrame  df has 21613 rows and 21 columns

###Summarize the total number of bedrooms in the dataset

In [6]:
describe(df[:bedrooms])

Summary Stats:
Mean:         3.370842
Minimum:      0.000000
1st Quartile: 3.000000
Median:       3.000000
3rd Quartile: 4.000000
Maximum:      33.000000


###Find the mean value of price in the dataset

In [7]:
mean(df[:price])

540088.1419053348

###Find the median value of sqft_living

In [8]:
median(df[:sqft_living])

1910.0

###Subset data for house with 2 bedrooms

In [9]:
twoBedroom_data = df[df[:bedrooms] .== 2,:]

2760x21 DataFrame
| Row  | id         | date              | price   | bedrooms | bathrooms |
|------|------------|-------------------|---------|----------|-----------|
| 1    | 5631500400 | "20150225T000000" | 180000  | 2        | 1.0       |
| 2    | 9212900260 | "20140527T000000" | 468000  | 2        | 1.0       |
| 3    | 16000397   | "20141205T000000" | 189000  | 2        | 1.0       |
| 4    | 8091400200 | "20140516T000000" | 252700  | 2        | 1.5       |
| 5    | 2426039314 | "20141201T000000" | 280000  | 2        | 1.5       |
| 6    | 3626039271 | "20150205T000000" | 585000  | 2        | 1.75      |
| 7    | 9418400240 | "20141028T000000" | 355000  | 2        | 1.0       |
| 8    | 1332700270 | "20140519T000000" | 215000  | 2        | 2.25      |
| 9    | 3869900162 | "20140904T000000" | 335000  | 2        | 1.75      |
| 10   | 3530510041 | "20140723T000000" | 188500  | 2        | 1.75      |
| 11   | 4060000240 | "20140623T000000" | 205425  | 2        | 1.0       |
⋮
| 2749 | 3277800823 | "20140820T000000" | 327000  | 2        | 2.0       |
| 2750 | 8835770330 | "20140819T000000" | 1057000 | 2        | 1.5       |
| 2751 | 476000017  | "20141003T000000" | 553000  | 2        | 2.0       |
| 2752 | 3438501329 | "20140520T000000" | 305000  | 2        | 2.5       |
| 2753 | 6021503705 | "20141015T000000" | 329000  | 2        | 2.5       |
| 2754 | 567000385  | "20140623T000000" | 362500  | 2        | 1.5       |
| 2755 | 2025049203 | "20140610T000000" | 399950  | 2        | 1.0       |
| 2756 | 2767604724 | "20141015T000000" | 505000  | 2        | 2.5       |
| 2757 | 2767600688 | "20141113T000000" | 414500  | 2        | 1.5       |
| 2758 | 1972201967 | "20141031T000000" | 520000  | 2        | 2.25      |
| 2759 | 1523300141 | "20140623T000000" | 402101  | 2        | 0.75      |
| 2760 | 1523300157 | "20141015T000000" | 325000  | 2        | 0.75      |

| Row  | sqft_living | sqft_lot | floors | waterfront | view | condition |
|------|-------------|----------|--------|------------|------|-----------|
| 1    | 770         | 10000    | 1.0    | 0          | 0    | 3         |
| 2    | 1160        | 6000     | 1.0    | 0          | 0    | 4         |
| 3    | 1200        | 9850     | 1.0    | 0          | 0    | 4         |
| 4    | 1070        | 9643     | 1.0    | 0          | 0    | 3         |
| 5    | 1190        | 1265     | 3.0    | 0          | 0    | 3         |
| 6    | 1980        | 8550     | 1.0    | 0          | 0    | 3         |
| 7    | 2020        | 6720     | 1.0    | 0          | 0    | 3         |
| 8    | 1610        | 2040     | 2.0    | 0          | 0    | 4         |
| 9    | 1030        | 1066     | 2.0    | 0          | 0    | 3         |
| 10   | 1240        | 2493     | 1.0    | 0          | 0    | 4         |
| 11   | 880         | 6780     | 1.0    | 0          | 0    | 4         |
⋮
| 2749 | 1490        | 1627     | 2.0    | 0          | 0    | 3         |
| 2750 | 2370        | 184231   | 2.0    | 0          | 0    | 3         |
| 2751 | 1400        | 1512     | 2.0    | 0          | 0    | 3         |
| 2752 | 1590        | 2656     | 2.0    | 0          | 0    | 3         |
| 2753 | 980         | 1020     | 3.0    | 0          | 0    | 3         |
| 2754 | 940         | 1768     | 2.0    | 0          | 0    | 3         |
| 2755 | 710         | 1157     | 2.0    | 0          | 0    | 4         |
| 2756 | 1430        | 1201     | 3.0    | 0          | 0    | 3         |
| 2757 | 1210        | 1278     | 2.0    | 0          | 0    | 3         |
| 2758 | 1530        | 981      | 3.0    | 0          | 0    | 3         |
| 2759 | 1020        | 1350     | 2.0    | 0          | 0    | 3         |
| 2760 | 1020        | 1076     | 2.0    | 0          | 0    | 3         |

| Row  | grade | sqft_above | sqft_basement | yr_built | yr_renovated |
|------|-------|------------|---------------|----------|--------------|
| 1    | 6     | 770        | 0             | 1933     | 0            |
| 2    | 7

In [10]:
size(twoBedroom_data)

(2760,21)

There are 2760 houses with 2 bedrooms

###Subset the data for house with 2 bedrooms and sqft_living more than 1000

In [11]:
twoBedrooms_1000_sqft = df[(df[:bedrooms].== 2).*(df[:sqft_living] .>= 1000),:]

1824x21 DataFrame
| Row  | id         | date              | price   | bedrooms | bathrooms |
|------|------------|-------------------|---------|----------|-----------|
| 1    | 9212900260 | "20140527T000000" | 468000  | 2        | 1.0       |
| 2    | 16000397   | "20141205T000000" | 189000  | 2        | 1.0       |
| 3    | 8091400200 | "20140516T000000" | 252700  | 2        | 1.5       |
| 4    | 2426039314 | "20141201T000000" | 280000  | 2        | 1.5       |
| 5    | 3626039271 | "20150205T000000" | 585000  | 2        | 1.75      |
| 6    | 9418400240 | "20141028T000000" | 355000  | 2        | 1.0       |
| 7    | 1332700270 | "20140519T000000" | 215000  | 2        | 2.25      |
| 8    | 3869900162 | "20140904T000000" | 335000  | 2        | 1.75      |
| 9    | 3530510041 | "20140723T000000" | 188500  | 2        | 1.75      |
| 10   | 3992700335 | "20140707T000000" | 382500  | 2        | 1.0       |
| 11   | 4232901525 | "20140627T000000" | 665000  | 2        | 1.0       |
⋮
| 1813 | 148000072  | "20140818T000000" | 600000  | 2        | 2.5       |
| 1814 | 7625702277 | "20150331T000000" | 406000  | 2        | 2.0       |
| 1815 | 8562780180 | "20140612T000000" | 336750  | 2        | 2.25      |
| 1816 | 3277800823 | "20140820T000000" | 327000  | 2        | 2.0       |
| 1817 | 8835770330 | "20140819T000000" | 1057000 | 2        | 1.5       |
| 1818 | 476000017  | "20141003T000000" | 553000  | 2        | 2.0       |
| 1819 | 3438501329 | "20140520T000000" | 305000  | 2        | 2.5       |
| 1820 | 2767604724 | "20141015T000000" | 505000  | 2        | 2.5       |
| 1821 | 2767600688 | "20141113T000000" | 414500  | 2        | 1.5       |
| 1822 | 1972201967 | "20141031T000000" | 520000  | 2        | 2.25      |
| 1823 | 1523300141 | "20140623T000000" | 402101  | 2        | 0.75      |
| 1824 | 1523300157 | "20141015T000000" | 325000  | 2        | 0.75      |

| Row  | sqft_living | sqft_lot | floors | waterfront | view | condition |
|------|-------------|----------|--------|------------|------|-----------|
| 1    | 1160        | 6000     | 1.0    | 0          | 0    | 4         |
| 2    | 1200        | 9850     | 1.0    | 0          | 0    | 4         |
| 3    | 1070        | 9643     | 1.0    | 0          | 0    | 3         |
| 4    | 1190        | 1265     | 3.0    | 0          | 0    | 3         |
| 5    | 1980        | 8550     | 1.0    | 0          | 0    | 3         |
| 6    | 2020        | 6720     | 1.0    | 0          | 0    | 3         |
| 7    | 1610        | 2040     | 2.0    | 0          | 0    | 4         |
| 8    | 1030        | 1066     | 2.0    | 0          | 0    | 3         |
| 9    | 1240        | 2493     | 1.0    | 0          | 0    | 4         |
| 10   | 1190        | 4440     | 1.0    | 0          | 0    | 3         |
| 11   | 1110        | 3200     | 1.0    | 0          | 0    | 3         |
⋮
| 1813 | 1830        | 1988     | 2.0    | 0          | 0    | 3         |
| 1814 | 1110        | 1095     | 3.0    | 0          | 0    | 3         |
| 1815 | 1170        | 1011     | 2.0    | 0          | 0    | 3         |
| 1816 | 1490        | 1627     | 2.0    | 0          | 0    | 3         |
| 1817 | 2370        | 184231   | 2.0    | 0          | 0    | 3         |
| 1818 | 1400        | 1512     | 2.0    | 0          | 0    | 3         |
| 1819 | 1590        | 2656     | 2.0    | 0          | 0    | 3         |
| 1820 | 1430        | 1201     | 3.0    | 0          | 0    | 3         |
| 1821 | 1210        | 1278     | 2.0    | 0          | 0    | 3         |
| 1822 | 1530        | 981      | 3.0    | 0          | 0    | 3         |
| 1823 | 1020        | 1350     | 2.0    | 0          | 0    | 3         |
| 1824 | 1020        | 1076     | 2.0    | 0          | 0    | 3         |

| Row  | grade | sqft_above | sqft_basement | yr_built | yr_renovated |
|------|-------|------------|---------------|----------|--------------|
| 1    | 7     | 860        | 300           | 1942     | 0            |
| 2    | 7

In [12]:
size(twoBedrooms_1000_sqft)

(1824,21)

There are 1824 houses with 2 bedrooms with sqft_living more than 1000 sqft

### Sort the dataset according to the increasing order of price 

In [13]:
sorted_data = sort(df,cols = :price)

21613x21 DataFrame
| Row   | id         | date              | price   | bedrooms | bathrooms |
|-------|------------|-------------------|---------|----------|-----------|
| 1     | 3421079032 | "20150217T000000" | 75000   | 1        | 0.0       |
| 2     | 40000362   | "20140506T000000" | 78000   | 2        | 1.0       |
| 3     | 8658300340 | "20140523T000000" | 80000   | 1        | 0.75      |
| 4     | 3028200080 | "20150324T000000" | 81000   | 2        | 1.0       |
| 5     | 3883800011 | "20141105T000000" | 82000   | 3        | 1.0       |
| 6     | 1623049041 | "20140508T000000" | 82500   | 2        | 1.0       |
| 7     | 7999600180 | "20140529T000000" | 83000   | 2        | 1.0       |
| 8     | 1523049188 | "20150430T000000" | 84000   | 2        | 1.0       |
| 9     | 2422049104 | "20140915T000000" | 85000   | 2        | 1.0       |
| 10    | 1322049150 | "20150305T000000" | 85000   | 2        | 1.0       |
| 11    | 1788900230 | "20140722T000000" | 86500   | 3        | 1.0       |
⋮
| 21602 | 6447300265 | "20141014T000000" | 4000000 | 4        | 5.5       |
| 21603 | 6065300370 | "20150506T000000" | 4208000 | 5        | 6.0       |
| 21604 | 3835500195 | "20140618T000000" | 4489000 | 4        | 3.0       |
| 21605 | 7738500731 | "20140815T000000" | 4500000 | 5        | 5.5       |
| 21606 | 1924059029 | "20140617T000000" | 4668000 | 5        | 6.75      |
| 21607 | 1247600105 | "20141020T000000" | 5110800 | 5        | 5.25      |
| 21608 | 7558700030 | "20150413T000000" | 5300000 | 6        | 6.0       |
| 21609 | 8907500070 | "20150413T000000" | 5350000 | 5        | 5.0       |
| 21610 | 2470100110 | "20140804T000000" | 5570000 | 5        | 5.75      |
| 21611 | 9208900037 | "20140919T000000" | 6885000 | 6        | 7.75      |
| 21612 | 9808700762 | "20140611T000000" | 7062500 | 5        | 4.5       |
| 21613 | 6762700020 | "20141013T000000" | 7700000 | 6        | 8.0       |

| Row   | sqft_living | sqft_lot | floors | waterfront | view | condition |
|-------|-------------|----------|--------|------------|------|-----------|
| 1     | 670         | 43377    | 1.0    | 0          | 0    | 3         |
| 2     | 780         | 16344    | 1.0    | 0          | 0    | 1         |
| 3     | 430         | 5050     | 1.0    | 0          | 0    | 2         |
| 4     | 730         | 9975     | 1.0    | 0          | 0    | 1         |
| 5     | 860         | 10426    | 1.0    | 0          | 0    | 3         |
| 6     | 520         | 22334    | 1.0    | 0          | 0    | 2         |
| 7     | 900         | 8580     | 1.0    | 0          | 0    | 3         |
| 8     | 700         | 20130    | 1.0    | 0          | 0    | 3         |
| 9     | 830         | 9000     | 1.0    | 0          | 0    | 3         |
| 10    | 910         | 9753     | 1.0    | 0          | 0    | 3         |
| 11    | 840         | 9480     | 1.0    | 0          | 0    | 3         |
⋮
| 21602 | 7080        | 16573    | 2.0    | 0          | 0    | 3         |
| 21603 | 7440        | 21540    | 2.0    | 0          | 0    | 3         |
| 21604 | 6430        | 27517    | 2.0    | 0          | 0    | 3         |
| 21605 | 6640        | 40014    | 2.0    | 1          | 4    | 3         |
| 21606 | 9640        | 13068    | 1.0    | 1          | 4    | 3         |
| 21607 | 8010        | 45517    | 2.0    | 1          | 4    | 3         |
| 21608 | 7390        | 24829    | 2.0    | 1          | 4    | 4         |
| 21609 | 8000        | 23985    | 2.0    | 0          | 4    | 3         |
| 21610 | 9200        | 35069    | 2.0    | 0          | 0    | 3         |
| 21611 | 9890        | 31374    | 2.0    | 0          | 4    | 3         |
| 21612 | 10040       | 37325    | 2.0    | 1          | 2    | 3         |
| 21613 | 12050       | 27600    | 2.5    | 0          | 3    | 4         |

| Row   | grade | sqft_above | sqft_basement | yr_built | yr_renovated |
|-------|-------|------------|---------------|----------|--------------|
| 1     | 3     | 670        

### Order the dataset in decreasing order of house prices

In [14]:
sorted_df_2 = sort(df,cols = :price, rev = true)

21613x21 DataFrame
| Row   | id         | date              | price   | bedrooms | bathrooms |
|-------|------------|-------------------|---------|----------|-----------|
| 1     | 6762700020 | "20141013T000000" | 7700000 | 6        | 8.0       |
| 2     | 9808700762 | "20140611T000000" | 7062500 | 5        | 4.5       |
| 3     | 9208900037 | "20140919T000000" | 6885000 | 6        | 7.75      |
| 4     | 2470100110 | "20140804T000000" | 5570000 | 5        | 5.75      |
| 5     | 8907500070 | "20150413T000000" | 5350000 | 5        | 5.0       |
| 6     | 7558700030 | "20150413T000000" | 5300000 | 6        | 6.0       |
| 7     | 1247600105 | "20141020T000000" | 5110800 | 5        | 5.25      |
| 8     | 1924059029 | "20140617T000000" | 4668000 | 5        | 6.75      |
| 9     | 7738500731 | "20140815T000000" | 4500000 | 5        | 5.5       |
| 10    | 3835500195 | "20140618T000000" | 4489000 | 4        | 3.0       |
| 11    | 6065300370 | "20150506T000000" | 4208000 | 5        | 6.0       |
⋮
| 21602 | 9320900420 | "20141014T000000" | 89000   | 3        | 1.0       |
| 21603 | 1788900230 | "20140722T000000" | 86500   | 3        | 1.0       |
| 21604 | 2422049104 | "20140915T000000" | 85000   | 2        | 1.0       |
| 21605 | 1322049150 | "20150305T000000" | 85000   | 2        | 1.0       |
| 21606 | 1523049188 | "20150430T000000" | 84000   | 2        | 1.0       |
| 21607 | 7999600180 | "20140529T000000" | 83000   | 2        | 1.0       |
| 21608 | 1623049041 | "20140508T000000" | 82500   | 2        | 1.0       |
| 21609 | 3883800011 | "20141105T000000" | 82000   | 3        | 1.0       |
| 21610 | 3028200080 | "20150324T000000" | 81000   | 2        | 1.0       |
| 21611 | 8658300340 | "20140523T000000" | 80000   | 1        | 0.75      |
| 21612 | 40000362   | "20140506T000000" | 78000   | 2        | 1.0       |
| 21613 | 3421079032 | "20150217T000000" | 75000   | 1        | 0.0       |

| Row   | sqft_living | sqft_lot | floors | waterfront | view | condition |
|-------|-------------|----------|--------|------------|------|-----------|
| 1     | 12050       | 27600    | 2.5    | 0          | 3    | 4         |
| 2     | 10040       | 37325    | 2.0    | 1          | 2    | 3         |
| 3     | 9890        | 31374    | 2.0    | 0          | 4    | 3         |
| 4     | 9200        | 35069    | 2.0    | 0          | 0    | 3         |
| 5     | 8000        | 23985    | 2.0    | 0          | 4    | 3         |
| 6     | 7390        | 24829    | 2.0    | 1          | 4    | 4         |
| 7     | 8010        | 45517    | 2.0    | 1          | 4    | 3         |
| 8     | 9640        | 13068    | 1.0    | 1          | 4    | 3         |
| 9     | 6640        | 40014    | 2.0    | 1          | 4    | 3         |
| 10    | 6430        | 27517    | 2.0    | 0          | 0    | 3         |
| 11    | 7440        | 21540    | 2.0    | 0          | 0    | 3         |
⋮
| 21602 | 900         | 4750     | 1.0    | 0          | 0    | 4         |
| 21603 | 840         | 9480     | 1.0    | 0          | 0    | 3         |
| 21604 | 830         | 9000     | 1.0    | 0          | 0    | 3         |
| 21605 | 910         | 9753     | 1.0    | 0          | 0    | 3         |
| 21606 | 700         | 20130    | 1.0    | 0          | 0    | 3         |
| 21607 | 900         | 8580     | 1.0    | 0          | 0    | 3         |
| 21608 | 520         | 22334    | 1.0    | 0          | 0    | 2         |
| 21609 | 860         | 10426    | 1.0    | 0          | 0    | 3         |
| 21610 | 730         | 9975     | 1.0    | 0          | 0    | 1         |
| 21611 | 430         | 5050     | 1.0    | 0          | 0    | 2         |
| 21612 | 780         | 16344    | 1.0    | 0          | 0    | 1         |
| 21613 | 670         | 43377    | 1.0    | 0          | 0    | 3         |

| Row   | grade | sqft_above | sqft_basement | yr_built | yr_renovated |
|-------|-------|------------|---------------|----------|--------------|
| 1     | 13    | 8570       

## Multi level sorting

### Sort house data according to the increasing order of bedrooms and price

In [15]:
Dual_levelsort = sort(df,cols = [:bedrooms,:price])

21613x21 DataFrame
| Row   | id         | date              | price   | bedrooms | bathrooms |
|-------|------------|-------------------|---------|----------|-----------|
| 1     | 9543000205 | "20150413T000000" | 139950  | 0        | 0.0       |
| 2     | 3980300371 | "20140926T000000" | 142000  | 0        | 0.0       |
| 3     | 6896300380 | "20141002T000000" | 228000  | 0        | 1.0       |
| 4     | 7849202190 | "20141223T000000" | 235000  | 0        | 0.0       |
| 5     | 2310060040 | "20140925T000000" | 240000  | 0        | 2.5       |
| 6     | 1222029077 | "20141029T000000" | 265000  | 0        | 0.75      |
| 7     | 1453602309 | "20140805T000000" | 288000  | 0        | 1.5       |
| 8     | 7849202299 | "20150218T000000" | 320000  | 0        | 2.5       |
| 9     | 2569500210 | "20141117T000000" | 339950  | 0        | 2.5       |
| 10    | 3374500520 | "20150429T000000" | 355000  | 0        | 0.0       |
| 11    | 3918400017 | "20150205T000000" | 380000  | 0        | 0.0       |
⋮
| 21602 | 1346300150 | "20141020T000000" | 3300000 | 8        | 4.0       |
| 21603 | 424049043  | "20140811T000000" | 450000  | 9        | 7.5       |
| 21604 | 1997200215 | "20140507T000000" | 599999  | 9        | 4.5       |
| 21605 | 2902200015 | "20150106T000000" | 700000  | 9        | 3.0       |
| 21606 | 8823901445 | "20150313T000000" | 934000  | 9        | 3.0       |
| 21607 | 9822700190 | "20140808T000000" | 1280000 | 9        | 4.5       |
| 21608 | 8823900290 | "20150317T000000" | 1400000 | 9        | 4.0       |
| 21609 | 5566100170 | "20141029T000000" | 650000  | 10       | 2.0       |
| 21610 | 8812401450 | "20141229T000000" | 660000  | 10       | 3.0       |
| 21611 | 627300145  | "20140814T000000" | 1148000 | 10       | 5.25      |
| 21612 | 1773100755 | "20140821T000000" | 520000  | 11       | 3.0       |
| 21613 | 2402100895 | "20140625T000000" | 640000  | 33       | 1.75      |

| Row   | sqft_living | sqft_lot | floors | waterfront | view | condition |
|-------|-------------|----------|--------|------------|------|-----------|
| 1     | 844         | 4269     | 1.0    | 0          | 0    | 4         |
| 2     | 290         | 20875    | 1.0    | 0          | 0    | 1         |
| 3     | 390         | 5900     | 1.0    | 0          | 0    | 2         |
| 4     | 1470        | 4800     | 2.0    | 0          | 0    | 3         |
| 5     | 1810        | 5669     | 2.0    | 0          | 0    | 3         |
| 6     | 384         | 213444   | 1.0    | 0          | 0    | 3         |
| 7     | 1430        | 1650     | 3.0    | 0          | 0    | 3         |
| 8     | 1490        | 7111     | 2.0    | 0          | 0    | 3         |
| 9     | 2290        | 8319     | 2.0    | 0          | 0    | 3         |
| 10    | 2460        | 8049     | 2.0    | 0          | 0    | 3         |
| 11    | 1470        | 979      | 3.0    | 0          | 2    | 3         |
⋮
| 21602 | 7710        | 11750    | 3.5    | 0          | 0    | 5         |
| 21603 | 4050        | 6504     | 2.0    | 0          | 0    | 3         |
| 21604 | 3830        | 6988     | 2.5    | 0          | 0    | 3         |
| 21605 | 3680        | 4400     | 2.0    | 0          | 0    | 3         |
| 21606 | 2820        | 4480     | 2.0    | 0          | 0    | 3         |
| 21607 | 3650        | 5000     | 2.0    | 0          | 0    | 3         |
| 21608 | 4620        | 5508     | 2.5    | 0          | 0    | 3         |
| 21609 | 3610        | 11914    | 2.0    | 0          | 0    | 4         |
| 21610 | 2920        | 3745     | 2.0    | 0          | 0    | 4         |
| 21611 | 4590        | 10920    | 1.0    | 0          | 2    | 3         |
| 21612 | 3000        | 4960     | 2.0    | 0          | 0    | 3         |
| 21613 | 1620        | 6000     | 1.0    | 0          | 0    | 5         |

| Row   | grade | sqft_above | sqft_basement | yr_built | yr_renovated |
|-------|-------|------------|---------------|----------|--------------|
| 1     | 7     | 844        

### Sort data in decreasing order of bedroom and increasing order of price

In [16]:
dual_sort_2 = sort(df,cols = [:bedrooms,:price],rev = [true,false])

21613x21 DataFrame
| Row   | id         | date              | price   | bedrooms | bathrooms |
|-------|------------|-------------------|---------|----------|-----------|
| 1     | 2402100895 | "20140625T000000" | 640000  | 33       | 1.75      |
| 2     | 1773100755 | "20140821T000000" | 520000  | 11       | 3.0       |
| 3     | 5566100170 | "20141029T000000" | 650000  | 10       | 2.0       |
| 4     | 8812401450 | "20141229T000000" | 660000  | 10       | 3.0       |
| 5     | 627300145  | "20140814T000000" | 1148000 | 10       | 5.25      |
| 6     | 424049043  | "20140811T000000" | 450000  | 9        | 7.5       |
| 7     | 1997200215 | "20140507T000000" | 599999  | 9        | 4.5       |
| 8     | 2902200015 | "20150106T000000" | 700000  | 9        | 3.0       |
| 9     | 8823901445 | "20150313T000000" | 934000  | 9        | 3.0       |
| 10    | 9822700190 | "20140808T000000" | 1280000 | 9        | 4.5       |
| 11    | 8823900290 | "20150317T000000" | 1400000 | 9        | 4.0       |
⋮
| 21602 | 3980300371 | "20140926T000000" | 142000  | 0        | 0.0       |
| 21603 | 6896300380 | "20141002T000000" | 228000  | 0        | 1.0       |
| 21604 | 7849202190 | "20141223T000000" | 235000  | 0        | 0.0       |
| 21605 | 2310060040 | "20140925T000000" | 240000  | 0        | 2.5       |
| 21606 | 1222029077 | "20141029T000000" | 265000  | 0        | 0.75      |
| 21607 | 1453602309 | "20140805T000000" | 288000  | 0        | 1.5       |
| 21608 | 7849202299 | "20150218T000000" | 320000  | 0        | 2.5       |
| 21609 | 2569500210 | "20141117T000000" | 339950  | 0        | 2.5       |
| 21610 | 3374500520 | "20150429T000000" | 355000  | 0        | 0.0       |
| 21611 | 3918400017 | "20150205T000000" | 380000  | 0        | 0.0       |
| 21612 | 6306400140 | "20140612T000000" | 1095000 | 0        | 0.0       |
| 21613 | 2954400190 | "20140624T000000" | 1295648 | 0        | 0.0       |

| Row   | sqft_living | sqft_lot | floors | waterfront | view | condition |
|-------|-------------|----------|--------|------------|------|-----------|
| 1     | 1620        | 6000     | 1.0    | 0          | 0    | 5         |
| 2     | 3000        | 4960     | 2.0    | 0          | 0    | 3         |
| 3     | 3610        | 11914    | 2.0    | 0          | 0    | 4         |
| 4     | 2920        | 3745     | 2.0    | 0          | 0    | 4         |
| 5     | 4590        | 10920    | 1.0    | 0          | 2    | 3         |
| 6     | 4050        | 6504     | 2.0    | 0          | 0    | 3         |
| 7     | 3830        | 6988     | 2.5    | 0          | 0    | 3         |
| 8     | 3680        | 4400     | 2.0    | 0          | 0    | 3         |
| 9     | 2820        | 4480     | 2.0    | 0          | 0    | 3         |
| 10    | 3650        | 5000     | 2.0    | 0          | 0    | 3         |
| 11    | 4620        | 5508     | 2.5    | 0          | 0    | 3         |
⋮
| 21602 | 290         | 20875    | 1.0    | 0          | 0    | 1         |
| 21603 | 390         | 5900     | 1.0    | 0          | 0    | 2         |
| 21604 | 1470        | 4800     | 2.0    | 0          | 0    | 3         |
| 21605 | 1810        | 5669     | 2.0    | 0          | 0    | 3         |
| 21606 | 384         | 213444   | 1.0    | 0          | 0    | 3         |
| 21607 | 1430        | 1650     | 3.0    | 0          | 0    | 3         |
| 21608 | 1490        | 7111     | 2.0    | 0          | 0    | 3         |
| 21609 | 2290        | 8319     | 2.0    | 0          | 0    | 3         |
| 21610 | 2460        | 8049     | 2.0    | 0          | 0    | 3         |
| 21611 | 1470        | 979      | 3.0    | 0          | 2    | 3         |
| 21612 | 3064        | 4764     | 3.5    | 0          | 2    | 3         |
| 21613 | 4810        | 28008    | 2.0    | 0          | 0    | 3         |

| Row   | grade | sqft_above | sqft_basement | yr_built | yr_renovated |
|-------|-------|------------|---------------|----------|--------------|
| 1     | 7     | 1040       